packages/clerk-js/src/ui/components/OrganizationSwitcher/OrganizationSwitcherTrigger.tsx

```diff
@@ -1,29 +1,71 @@
-import { forwardRef } from 'react';
+import { forwardRef, useCallback } from 'react';
 
-import { useCoreOrganization, useCoreUser, useOrganizationSwitcherContext } from '../../contexts';
-import { Button, descriptors, Icon, localizationKeys } from '../../customizables';
+import {
+  useCoreOrganization,
+  useCoreOrganizationList,
+  useCoreUser,
+  useOrganizationSwitcherContext,
+} from '../../contexts';
+import { Box, Button, descriptors, Icon, localizationKeys, NotificationBadge } from '../../customizables';
 import { OrganizationPreview, PersonalWorkspacePreview, withAvatarShimmer } from '../../elements';
+import { useDelayedVisibility, usePrefersReducedMotion } from '../../hooks';
 import { Selector } from '../../icons';
-import type { PropsOfComponent } from '../../styledSystem';
+import type { PropsOfComponent, ThemableCssProp } from '../../styledSystem';
+import { animations } from '../../styledSystem';
+import { organizationListParams } from './utils';
 
 type OrganizationSwitcherTriggerProps = PropsOfComponent<typeof Button> & {
   isOpen: boolean;
 };
 
+function useEnterAnimation() {
+  const prefersReducedMotion = usePrefersReducedMotion();
+
+  const getFormTextAnimation = useCallback(
```

Comment:
This helper looks more complicated than I'd expect. 
Do we need the useCallback here? Is there any performance issue or side-effect if we don't wrap the handler with it? In general, let's prefer not using useCallback or useMemo unless needed. 

If we manage to remove useCallback, I'd probably inline this behavior inside a different `NotificationCountBadge` component that would include the animation logic as well

Reply:
I removed useCallback. I created NotificationCountBadge but it is still nested in OrgSwitcher. Designs still don't indicate that this will be used globally or that the animation will always be used

Reply:
Thats great :)